### PR TITLE
[Rewrite Branch] Add debounce of 2 seconds to analytics for note edit

### PR DESCRIPTION
**Pull request by belcherj**
### Fix

Current production has a debounce of 2 seconds on `editor_note_edited`. Additionally, it could have a performance impact sending analytics on every key press. I was not able to see any performance impact on staging.

Note we are not sending what was written only that an edit was made.

### Test
1. Analytics does not run on dev so you will have to comment out: https://github.com/Automattic/simplenote-electron/blob/develop/lib/analytics/index.ts#L58 to test locally
2. Turn on analytics
3. Open Chrome dev tools to the network panel
4. Type something in a note
5. Notice there was a 2-second delay before network call gets sent

### CODE REVIEW

**Review by codebykat — APPROVED** [review]
Thank you!!

File: lib/state/analytics/middleware.ts
Comment (dmsnell) [inline lib/state/analytics/middleware.ts]: we don't have any hard rules in this repo but in most Automattic JS code we put external exports at the top. would be nice if we could move this before merging but it's not a blocker.

File: lib/state/analytics/middleware.ts
Comment (dmsnell) [inline lib/state/analytics/middleware.ts:40]: nitpick: we need a blank line above this.

**Review by dmsnell — APPROVED** [review]
There are a couple things I think it would be worth updating here but it's good to merge after that/otherwise. Thanks for adding this in and preventing our stats from surging artificially.

**Comment by codebykat** [comment]
Nits picked

**Review by dmsnell — APPROVED** [review]
Ship it!

**Pull request merged**
